api/v1beta1/openstackdataplanerole_types.go

```diff
@@ -107,12 +107,22 @@ func (instance *OpenStackDataPlaneRole) InitConditions() {
 		condition.UnknownCondition(condition.DeploymentReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(condition.InputReadyCondition, condition.InitReason, condition.InitReason),
 		condition.UnknownCondition(SetupReadyCondition, condition.InitReason, condition.InitReason),
-		condition.UnknownCondition(RoleBareMetalProvisionReadyCondition, condition.InitReason, condition.InitReason),
-		condition.UnknownCondition(RoleIPReservationReadyCondition, condition.InitReason, condition.InitReason),
-		condition.UnknownCondition(RoleDNSDataReadyCondition, condition.InitReason, condition.InitReason),
 	)
 
-	if instance.Spec.Services != nil {
+	// Only set Baremetal related conditions if we have baremetal hosts included in the
+	// baremetalSetTemplate.
+	if len(instance.Spec.BaremetalSetTemplate.BaremetalHosts) > 0 {
+		bmConditionsList := []condition.Type{
+			RoleBareMetalProvisionReadyCondition,
+			RoleIPReservationReadyCondition,
```

Comment:
This is wrong too, we should not remove  these conditions. The earlier code was correct.